lib-cloudsdk/src/main/java/com/facebook/openwifi/cloudsdk/UCentralUtils.java

```diff
@@ -422,20 +482,61 @@ public static Map<String, String> getBssidsMap(
 	 * Determines if the given channel is in the given band.
 	 *
 	 * @param channel channel number
-	 * @param band    "2G" or "5G"
+	 * @param band band (e.g., "2G")
 	 * @return true if the given channel is in the given band; false otherwise
 	 */
 	public static boolean isChannelInBand(int channel, String band) {
-		return AVAILABLE_CHANNELS_BAND.get(band).contains(channel);
+		List<Integer> channels = AVAILABLE_CHANNELS_BAND.get(band);
+		return channels.get(0) <= channel &&
+			channel <= channels.get(channels.size() - 1);
 	}
 
-	/** Return which band contains the given frequency (MHz). */
-	public static String freqToBand(int freqMHz) {
-		if (2412 <= freqMHz && freqMHz <= 2484) {
-			return "2G";
+	/**
+	 * Converts channel number to that channel's center frequency (MHz) in the
+	 * given band.
+	 *
+	 * @param band band (e.g., "2G")
+	 * @param channel channel number
+	 * @return the center frequency of the given channel in MHz
+	 */
+	public static int channelToFrequencyMHz(String band, int channel) {
+		if (band.equals(UCentralConstants.BAND_2G)) {
+			if (channel == 14) {
+				// special case
+				return 2484;
+			} else {
+				return 2407 + 5 * channel;
+			}
+		} else if (band.equals(UCentralConstants.BAND_5G)) {
+			return 5000 + 5 * channel;
 		} else {
-			return "5G";
+			// 6G
+			return 5950 + 5 * channel;
+		}
+	}
+
+	/**
+	 * Return the band that contains the given frequency (MHz).
+	 *
+	 * @param freqMHz center frequency (MHz)
+	 * @return the band that contains the given frequency, or null if no band
+	 *         contains it.
+	 */
+	public static String freqToBand(int freqMHz) {
```

Comment:
not important now -- but just compute and store this as class state, if this method is frequently called